### DISCUSSION TITLE
Function calls from templates are a performance killer

### DISCUSSION THREAD

**Discussion by frankanneveld** [primefaces/community#1737] [PrimeNG] [answered]
I'm new here, and for a new project, we have to compare different UI libraries. So I started diving into primeNg and clicked around, and then I saw that Angular uses the new template flow, but I can imagine it is not there because it's very new. But in the same file ([https://github.com/primefaces/primeng/blob/master/src/app/components/menubar/menubar.ts#L143](https://github.com/primefaces/primeng/blob/master/src/app/components/menubar/menubar.ts#L143)) I saw a lot of function calls from the template to the component. As we all know, this dramatically slows down the performance every time Angular changes its cycle to determine the new value. All the functions in the component are evaluated again. It is not a bug, but it would be nice, performance-wise, to have a better solution, or maybe Signals can be a solution. What would you think?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We want to use new features of Angular too but it's not easy to implement new features to +90 components without breaking anything. So we need to be careful. We want to adapt new features of Angular but it happens slowly because we have to think +400k users. We aim to make the library more stable in v18 while adapting new Angular features.

Thanks a lot for the valuable feedback.

**Reply by frankanneveld** [reply]
Hi Cetincakiroglu,

Thank you for your reply. I can understand; it is indeed not easy and quite a task. Angular also made a lot of changes. I have to do some refactoring on my projects too. Looking forward to version 18, I appreciate the update on this.  👍

Greetings,
Frank
